### DISCUSSION TITLE
Viewing source code of libraries (like Spring) from nvim

### DISCUSSION THREAD

**Discussion by MorKaragh** [mfussenegger/nvim-jdtls#668] [Q&A]
Hello!

I have set up everything I need in neovim + jdtls, but I struggle with a single issue and I hope you can help me.

Sometimes we need to see the source code of dependencies added with maven, like Spring classes or something else. 
In Intellij Idea it is quite easy, but in neovim I don't understand how to make it work. 
I have tried several different configs from the internet and none of them seems to be working in that way. 

Therefore, I have a question - what neovim should to to get me to the source code of a dependency?
Obviously, it has to get this code somewhere. Where? Should I download it manually and somehow connect to my project?
Or, maybe maven is able to do it and I need to look closer into integration of the jdtls with maven?

I hope that if you can give me the general idea of how this should work I'll figure the rest by myself. 
Example configs that have that feature could be also very helpful.

I do not attach my config here because now I use the most basic one, as it gives me almost everything I need for now (I don't use debugger or lombok). But without being able to read dependency sources I will have to use Idea and I don't want to. 

Please, help.

**Comment by MorKaragh** [comment]
Well, looks like I have done something wrong, but not sure what exactly. 
Now it works fine. This project helped me to figure out the correct setup: https://github.com/bcampolo/nvim-starter-kit/tree/java#neovim-starter-kit-for-java-

**Comment by rogeriofrsouza** [comment]
I think this is the code needed in your LSP settings.
```lua
settings = {
    java = {
      eclipse = {
        downloadSources = true,
      },
      maven = {
        downloadSources = true,
      },
  },
  ```
  Other than that, maybe you need to run a mvn command to download dependencies sources, like:
  `mvn dependency:sources dependency:resolve -Dclassifier=javadoc`
